### PR TITLE
[Quest] Acting in Good Faith - random qm1 spawn on 4 possible braziers

### PR DESCRIPTION
**Pull request by jamesbradleym**
**_I affirm:_**
- [x] I understand that if I do not agree to the following points by completing the checkboxes my PR will be ignored.
- [x] I understand I should leave resolving conversations to the LandSandBoat team so that reviewers won't miss what was said.
- [x] I have read and understood the [Contributing Guide](https://github.com/LandSandBoat/server/blob/base/CONTRIBUTING.md) and the [Code of Conduct](https://github.com/LandSandBoat/server/blob/base/CODE_OF_CONDUCT.md).
- [x] I have _**tested my code and the things my code has changed**_ since the last commit in the PR and will test after any later commits.

## What does this pull request do?

Updates Acting in Good Faith qm1 to randomly spawn between 4 valid brazier positions

## Steps to test these changes

Zone in to The Eldieme Necropolis and see ??? on one of the braziers at:

{ -460.85, -1.5, 425.14 }, -- D-5.
{ -24.10, -9.303, 258.993 }, -- I-7.
{ -19.624, -1.631, 60.368 }, -- I-10.
{ 256.757, -20.489, 335.920 }, -- M-6.

Valid brazier location sources:
https://wiki.ffo.jp/html/10507.html
https://ffxiclopedia.fandom.com/wiki/Acting_in_Good_Faith
https://static.wikia.nocookie.net/ffxi/images/2/20/The_Eldieme.png/revision/latest?cb=20080806210806

### CODE REVIEW

File: scripts/zones/The_Eldieme_Necropolis/IDs.lua [hunk 1]
```diff
@@ -79,6 +81,15 @@ zones[xi.zone.THE_ELDIEME_NECROPOLIS] =
         TREASURE_COFFER    = GetFirstID('Treasure_Coffer'),
         SARCOPHAGUS_OFFSET = GetFirstID('Sarcophagus'),
         CANDLE_OFFSET      = GetFirstID('_5fu'),
+        qm1                = GetFirstID('qm1'),
```
Comment (Xaver-DaRed) [inline scripts/zones/The_Eldieme_Necropolis/IDs.lua]: Enums in all caps

File: scripts/quests/windurst/Acting_in_Good_Faith.lua [hunk 2]
```diff
@@ -86,8 +89,10 @@ quest.sections =
             onEventFinish =
             {
                 [50] = function(player, csid, option, npc)
-                    player:messageSpecial(eldiemeID.text.SPIRIT_INCENSE_EMITS_PUTRID_ODOR, xi.ki.SPIRIT_INCENSE)
+                    player:messageSpecial(ID.text.SPIRIT_INCENSE_EMITS_PUTRID_ODOR, xi.ki.SPIRIT_INCENSE)
                     player:delKeyItem(xi.ki.SPIRIT_INCENSE)
+
+                    npc:setPos(unpack(ID.npc.QM1_POS[math.random(1, 4)])) -- move qm1 randomly on quest completion
```
Comment (Xaver-DaRed) [inline scripts/quests/windurst/Acting_in_Good_Faith.lua]: `ID` references nothing. use the already existing `eldiemeID`

**Comment by Xaver-DaRed** [comment]
Where did you get this positios from?

**Comment by jamesbradleym** [comment]
> Where did you get this positios from?

I went to the brazier locations in game and got my position w/ wallhack on 😅

File: scripts/quests/windurst/Acting_in_Good_Faith.lua [hunk 3]
```diff
@@ -74,7 +77,7 @@ quest.sections =
 
         [xi.zone.THE_ELDIEME_NECROPOLIS] =
         {
-            ['qm1'] =
+            ['QM1'] =
```
Comment (Xaver-DaRed) [inline scripts/quests/windurst/Acting_in_Good_Faith.lua]: this is not an enum. its the actual npc name. revert the name to how it was, please.

Also, remember to test your changes

File: scripts/quests/windurst/Acting_in_Good_Faith.lua (hunk 2 above)
Comment (jamesbradleym) [reply scripts/quests/windurst/Acting_in_Good_Faith.lua]: Switched `eldiemeID` to just `ID` if that's alright 😬

Comment (Xaver-DaRed) [reply scripts/quests/windurst/Acting_in_Good_Faith.lua]: I'm sorry, but its not. revert it to how it was.

File: scripts/quests/windurst/Acting_in_Good_Faith.lua (hunk 3 above)
Comment (jamesbradleym) [reply scripts/quests/windurst/Acting_in_Good_Faith.lua]: Got too trigger happy when hunting for qm1 to update 😔

File: scripts/quests/windurst/Acting_in_Good_Faith.lua (hunk 2 above)
Comment (jamesbradleym) [reply scripts/quests/windurst/Acting_in_Good_Faith.lua]: Done ✅

**Review by Xaver-DaRed — APPROVED** [review]

**Comment by Xaver-DaRed** [comment]
Could you rebase this, please, so the new CI check can pass?

**Review by zach2good — APPROVED** [review]

**Pull request merged**